Recovers separate output dirs for different forcefields

j3mdamas opened the pull request:
Also:
- changes in printing functions
- fixing parameterize tutorial
- minor fixes in long tests

This won't pass tests for now, as I haven't changed the data. @raimis, can you pre-approve what's done, and then I change data to match the changes?

raimis commented on htmd/parameterization/cli.py:
```diff
@@ -481,14 +481,14 @@ def _printEnergies(molecule, parameters, filename):
-Improper : {IMPROPER_ENERGY}
-Electro  : {ELEC_ENERGY}
-VdW      : {VDW_ENERGY}
+Bond     : {BOND_ENERGY:10.3f}
```
We need `.6f` as our test tolerances are set to 1e-6

raimis commented on htmd/parameterization/cli.py:
```diff
@@ -481,14 +481,14 @@ def _printEnergies(molecule, parameters, filename):
 
     string = '''
-== Diagnostic Energies ==
+== Diagnostic Energies (in kcal/mol) ==
```
Units at the end of the quantity, not in the header

j3mdamas replied:
Any scientific paper disagrees with you, but as you wish

raimis replied:
Just keeping consistent with the other log messages

raimis commented on htmd/parameterization/cli.py:
```diff
@@ -497,13 +497,29 @@ def _printEnergies(molecule, parameters, filename):
            ELEC_ENERGY=energies['elec'],
            VDW_ENERGY=energies['vdw'])
 
+    for l in string.split('\n'):
```
Human-readable variable name: `l` --> `line`

j3mdamas replied:
Done.

raimis commented on htmd/parameterization/cli.py:
```diff
@@ -497,13 +497,29 @@ def _printEnergies(molecule, parameters, filename):
-    for name in ('bond', 'angle', 'dihedral', 'improper', 'elec', 'vdw'):
-        logger.info('   {:8s} : {:10.3f} kcal/mol'.format(name, energies[name]))
+
+def _printArguments(args, filename=None):
```
Can the function definition be moved  before, `_prepare_molecule` in the file.

j3mdamas replied:
Sure, no problem.

raimis commented on htmd/parameterization/cli.py:
```diff
@@ -497,13 +497,29 @@ def _printEnergies(molecule, parameters, filename):
+
+def _printArguments(args, filename=None):
+
+    if filename:
```
Looks a bit over-engineered way to write a text file.

j3mdamas replied:
I've added some comments to make it more obvious

raimis commented on htmd/parameterization/cli.py:
```diff
@@ -512,8 +528,13 @@ def _output_results(mol, parameters, original_coords, args):
 
     logger.info('=== Results ===')
 
+    dir = os.path.join(args.outdir, 'parameters', args.forcefield)
```
`dir` --> `paramDir`. It shadow built-in `dir`

j3mdamas replied:
Correct!

raimis commented on htmd/parameterization/cli.py:
```diff
@@ -549,12 +570,7 @@ def main_parameterize(arguments=None):
-        if key in ('fake_qm',):  # Hidden
-            continue
-        logger.info('{:>20s}: {:s}'.format(key, str(value)))        
+    _printArguments(args)
```
This should go after the parse.

j3mdamas replied:
Done

j3mdamas commented:
I've just addressed your comments. Let me know if it's OK so I can change the data accordingly.

j3mdamas commented:
New data added: https://github.com/Acellera/htmd-data/pull/31

Some fixes to the tests (particular notice to the adding of a new assertion to overcome a "bug/feature" of `zip`). Fixed also some bugs of the cli that were overseen in my previous single test.

If it passes travis, I think it's ready to merge, but feel free to review the fixes I introduced.

@raimis

raimis commented:
@j3mdamas Great work! Thanks for fixing all the tests.
